lib/ruby_parser_extras.rb

```diff
@@ -880,6 +880,77 @@ def new_string val
     result
   end
 
+  def new_qword_list_entry val
+    str = val[1]
+    str.force_encoding("ASCII-8BIT") unless str.valid_encoding? unless RUBY_VERSION < "1.9"
+    result = s(:str, str)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_qword_list
+    result = s(:array)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_word_list
+    result = s(:array)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_word_list_entry val
+    result = val[1][0] == :evstr ? s(:dstr, "", val[1]) : val[1]
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_qsym_list
+    result = s(:array)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_qsym_list_entry val
+    result = s(:lit, val[1].to_sym)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_symbol_list
+    result = s(:array)
+    self.lexer.lineno += self.lexer.extra_lineno
+    self.lexer.extra_lineno = 0
+    result
+  end
+
+  def new_symbol_list_entry val
+    list, sym, _ = val
+    result = val[1]
+
+    result ||= s(:str, "")
+
+    case sym[0]
+    when :dstr then
+      sym[0] = :dsym
+    when :str then
+      sym = s(:lit, sym.last.to_sym)
+    else
+      debug20 24
+      sym = s(:dsym, "", sym || s(:str, ""))
```

Comment:
Here is the replacement for that line that was using `result` above. I looked around at other similar code you had in that file and replaced `result` with `sym || s(:str, "")`. Let me know if that is gibberish or not.